tutorials/diffusion_calculation.ipynb

```diff
@@ -0,0 +1,293 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": true,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Diffusion mapping tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Start by importing the required libraries and defining some settings:"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "import os\n",
+    "\n",
+    "from ukat.data import fetch\n",
+    "from ukat.mapping.diffusion import ADC, DTI\n",
+    "\n",
+    "# Ensure figures are rendered in the notebook\n",
+    "%matplotlib inline"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Tell python where to store the maps NIfTI files we will generate. We will store it in a directory called\n",
+    "`diffusion_calculation_output` in the `tutorials` directory."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "outputs": [],
+   "source": [
+    "# Initialise output path for diffusion maps\n",
+    "directory = 'diffusion_calculation_output'\n",
+    "os.makedirs(directory, exist_ok=True)\n",
+    "OUTPUT_DIR = os.path.join(os.getcwd(), directory)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Retrieve diffusion test data using our data fetcher.\n",
+    "This will return 3 variables:\n",
+    "* `image`: a 4D array with the magnitude data at all echo times.\n",
```

Comment:
Shouldn't it be bvalues/bvecs instead of echo times? There are different echo times indeed. Let me know if I'm wrong here please.

Reply:
Correct, that's a copy paste error.